### ISSUE TITLE
Make the ESLint config more DRY

### ISSUE THREAD

**Issue by silvenon** [kiwicom/orbit#2347]
Currently our ESLint config for JavaScript (Flow) and TypeScript are strictly separated, and there is not only a lot of repetition, but even differences between the two. We should extract the common parts of the configuration and `eslintJS.js` and `eslintTS.js` should strictly contain config related to Flow and TypeScript respectively, while everything else should be in `eslintCommon.js` that both would extend.

**Comment by yencolon** [comment]
Hi, I think I can help you with this!

**Comment by silvenon** [comment]
That would be very much appreciated! Let me know if you need any help 😉 The configs have most likely diverged significantly, so when in doubt, prefer the rules that are in `eslintJS.js`.

**Comment by yencolon** [comment]
Of course, assign this to me 👍

**Comment by yencolon** [comment]
Hi @silvenon, I'm currently working on this, but I am a little bit confused, the issue description says that there's 'a lot of repetition' in both files, but I don't see it. Maybe I did not get the main idea. Can you help me?

**Comment by silvenon** [comment]
That's true, it's kinda standalone currently, but try applying to it everything that isn't specifically Flow-related so we see what happens. When we start adding TypeScript files at some point in the future, we want to have the same ESLint support as we did for Flow files.
